packages/ai/src/agent/do-stream-step.ts

```diff
@@ -365,5 +377,104 @@ export async function doStreamStep(
   //   throw new Error('LLM stream ended without a "finish" chunk');
   // }
 
-  return { toolCalls, finish };
+  const step = chunksToStep(chunks, toolCalls, conversationPrompt, finish);
+  return { toolCalls, finish, step };
+}
+
+function chunksToStep(
```

Comment:
Not sure if this whole function is necessary - we only want to get the "step" out so we can execute stepConditions on it, but technically, stepConditions can be passed in for arbitrary checks on steps, so it makes sense to synthesize the whole step here.
However, I imagine this logic must already exist somewhere in the AI SDK code? It feels odd to have to re-do it here, but I couldn't find a trivial way to extract this from the SDK when using model.doStream